test/helpers/deploy/deploy_resources.go

```diff
@@ -269,6 +269,10 @@ func KonnectGatewayControlPlaneWithID(
 		},
 	}
 	cp.Status.ID = uuid.NewString()[:8]
+	cp.Status.Endpoints = &konnectv1alpha1.KonnectEndpoints{
+		ControlPlaneEndpoint: "cp.endpoint",
+		TelemetryEndpoint:    "tp.endpoint",
+	}
```

Comment:
I believe we can add using the provided options here (before `Update()`) like so:

```suggestion
	cp.Status.Endpoints = &konnectv1alpha1.KonnectEndpoints{
		ControlPlaneEndpoint: "cp.endpoint",
		TelemetryEndpoint:    "tp.endpoint",
	}
	for _, opt := range opts {
		opt(cp)
	}
```

So that the call site at `test/envtest/dataplane_konnectextensions_test.go` becomes:

```
	cp := deploy.KonnectGatewayControlPlaneWithID(t, ctx, cl, konnectAPIAuthConfiguration, WithKonnectID(konnectControlPlaneID))
	require.NoError(t, cl.Status().Update(ctx, cp))
```